packages/data-explorer/src/views/MainView.vue

```diff
@@ -139,21 +139,23 @@ export default Vue.extend({
     this.fetchTablePermissions({ tableName })
     await this.fetchTableMeta({ tableName })
     this.setRouteQuery(this.$route.query)
-    this.setDataDisplayLayout(this.$route.params.view)
+    this.setDataDisplayLayout(this.$route.query.view)
     this.fetchViewData()
   },
   destroyed () {
     this.$eventBus.$off('delete-item')
   },
   async beforeRouteUpdate (to, from, next) {
+    this.setLoading(true)
```

Comment:
Is this beter here that in the actions?

Reply:
I think so! :) By putting the loading indicator here, its easier to spot where the loading occurs and where it ends.